ground/src/main/java/com/google/android/ground/persistence/remote/firestore/schema/LoiConverter.kt

```diff
@@ -35,15 +34,19 @@ object LoiConverter {
   const val GEOMETRY_COORDINATES = "coordinates"
   const val GEOMETRY = "geometry"
 
-  fun toLoi(survey: Survey, doc: DocumentSnapshot): Result<LocationOfInterest> {
+  fun toLoi(survey: Survey, doc: DocumentSnapshot): Result<LocationOfInterest> =
+    runCatching {
+      toLoiUnchecked(survey, doc)
+    }
+
+  private fun toLoiUnchecked(survey: Survey, doc: DocumentSnapshot): LocationOfInterest {
```

Comment:
I'm trying to separate where we use Exceptions vs Result to capture error states. This example handles downstream exceptions as well as error states generated here.

Reply:
Oh nice I didn't realize there was a runCatching we could use here that returned a Result.